docs-gh-pages/consensus.md

```diff
@@ -0,0 +1,91 @@
+## Consensus
+
+### Architecture
+Every new `CheckpointBlock` is being created under a consensus round.
+
+### Glossary
+- **consensus owner** - the node which sent consensus proposal to others and initiated a new round
+- **facilitator** - node which has been selected to take part in a new round and responded successfully to participate in it (including owner)
+- **SOE** - signed observation edge
+- **data proposal** - transactions (and other checkpoint block data) proposed by the facilitator to include in the new consensus round
+
+### Consensus round
+1. Start consensus
+2. Broadcast and gather consensus data proposals
+3. Merge, broadcast and gather consensus block proposals
+4. Select majority block proposal and merge facilitators' signatures
+5. Accepted finished block and broadcast to non-facilitators
+
+### Stages
+
+#### 1. Start consensus
+`ConsensusScheduler` runs `consensusManager.startOwnConsensus()` every **n** seconds
+(*default value: 10s*, configurable under `constellation.consensus.start-own-interval`).
+To start a new round, node must be in one of valid node states. Otherwise, creating new round will be skipped.
+Valid node states for starting new consensus round are **Ready** and **SnapshotCreation**.
+
+When new round can be created, owner creates one using pending transactions (up to `maxTransactionThreshold`), tip SOE
+and selects final facilitators. Peers which can take part in the consensus round must have **nextSnapshotHeight** above
+the proposed tip. Otherwise, the selected node will not be able to accept the final checkpoint block. Facilitators are being
+selected randomly taking into account above preconditions.
+
+**Dummy transaction** is a transaction pulled to the consensus round if node has no real transactions. Node should always
+be able to create a block, even if the block is empty. To distinguish empty blocks, there must be at least one dummy transaction
+which is equivalent to the real transaction but carries 0 amount.
+
+#### 2. Broadcast and gather consensus data proposals
+Owner sends the `NotifyFacilitators` message to all selected facilitators and waits for the response. If facilitator is able
+to start a consensus, responds with success message and creates a corresponding round. Facilitator cannot start a consensus
+if proposed tip is above the current snapshot height or is in the incorrect state (i.e. leaving from the cluster).
+When all facilitators have been notified and started rounds successfully, each one pulls transactions, similarly as owner did,
+and broadcasts it to all consensus round's participants. In parallel, each participant gathers proposals from others and keeps it locally.
+
+#### 3. Merge, broadcast and gather consensus block proposals
+When node received all proposals, it merges them together to create a block proposal and then broadcasts it to others.
```

Comment:
We could mention somewhere that the logic for merging proposals into a block proposal is deterministic and the same for all the nodes, to emphasize that for the same data we should always get the same block.